services/datamanager/builtin/builtin.go

```diff
@@ -512,16 +494,46 @@ func (svc *builtIn) uploadData(cancelCtx context.Context, intervalMins float64)
 			case <-ticker.C:
 				svc.lock.Lock()
 				if svc.syncer != nil {
-					svc.flushCollectors()
-					svc.syncer.SyncDirectory(svc.captureDir)
-					svc.syncAdditionalSyncPaths()
+					svc.sync()
 				}
 				svc.lock.Unlock()
 			}
 		}
 	})
 }
 
+func (svc *builtIn) sync() {
+	svc.flushCollectors()
```

Comment:
Good refactor: we're just syncing a bunch of files, of which arbitrary files are a subset. If this expands in the future, makes it a lot easier to update in one place.